src/test/java/org/springframework/data/elasticsearch/core/MappingBuilderTests.java

```diff
@@ -249,6 +252,33 @@ public void shouldMapBooks() {
 		// then
 	}
 
+	@Test // DATAES-618
+	public void shouldUseDocValues() {
+
```

Comment:
to check for the mapping you do not need to create an index, put and retrieve a mapping. Check the `shouldUseFieldNameOnId()` method for example, how it is done there:

define the expected JSON string (no need to format it), build the mapping and then compare it with `JSONAssert.assertEquals(...)`

Reply:
By putting mapping and getting mapping, it is covering the integration test also. If required I will include additional test cases with your suggested model.

Reply:
` MappingBuilderTests` is a test class to unit test the building of the mappings. I know there are tests in that class that use the template and the index (old ones), but as you said, these are integration tests, and I don't think that the `MappingBuilder` should be tested that way.
The `MappingBuilder`'s task is to create a JSON string from a `@Document` class, and that should be tested.

Reply:
we should have tests that check that a doc_value=true is not written to the mapping, no matter if coming from the default, of being set explicitly.
What about the special types like GeoPoint? If they cannot have a doc_type we should test that no one can be set, if the can have it, we must test, that it is written to the mapping. Same for all possible fields thatz may not be handled by the `addFieldMappingParameters` method.